static/css/search.css

```diff
@@ -1,16 +1,75 @@
-/* CSS related to customing the Algolia Search bar */
+/* CSS related to customizing the Algolia DocSearch v4 search bar */
 
-.DocSearch-Button {
-    width: -webkit-fill-available;
-    margin: 10px !important;
-}
-
-.search-404-wrapper .DocSearch-Button {
-    margin: 0px !important;
+/* Custom color variables for balena branding */
+:root {
+    --docsearch-primary-color: #00aeef;
+    --docsearch-text-color: #1f2937;
+    --docsearch-spacing: 12px;
+    --docsearch-highlight-color: #00aeef;
+    --docsearch-muted-color: #6b7280;
+    --docsearch-container-background: rgba(0, 0, 0, 0.6);
+    --docsearch-modal-background: #ffffff;
 }
 
+/* Search wrapper container */
 .search-wrapper {
     border-bottom: 1px solid #e6e8eb;
     border-right: 1px solid #e6e8eb;
     background: #f7f7f7;
+    width: 100%;
+    padding: 10px;
+    box-sizing: border-box;
+}
+
+/* Main search button - full width in sidebar - Override DocSearch v4 'all' property */
+.DocSearch-Button {
+    all: unset !important;
+    width: 100% !important;
+    max-width: none !important;
+    margin: 0 !important;
+    box-sizing: border-box !important;
+    display: flex !important;
+    align-items: center !important;
+    justify-content: space-between !important;
+    padding: 0 12px !important;
+    height: 44px !important;
+    background: #ffffff !important;
+    border: 1px solid #d0d5dd !important;
+    border-radius: 6px !important;
+    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.05) !important;
+    transition: all 0.15s ease-in-out !important;
+    cursor: pointer !important;
+}
```

Comment:
you can remove the need for `!important` by increasing the specificity of the selector, something like

`.sidebar .DocSearch-Button {`

should work.